### ISSUE TITLE
Support Emojis

### ISSUE THREAD

**Issue by finnp** [mikaelbr/marked-terminal#14]
GitHub flavored markdown allows for emojis like this `:poop:` :poop:. We should support them as well. Haven't found a good module yet for the conversion though.

**Comment by mikaelbr** [comment]
That would be cool. Mac Terminal supports them at least, I don't know about cross platform though. For instance what does

```shell
$ echo "\xF0\x9F\x98\x80"
```

show on Linux or Windows?

Github has a overview of all the emojis it support and it's unicode here: https://api.github.com/emojis

**Comment by mikaelbr** [comment]
I have a working example of this now, using something like:

```js
function insertEmojis(text) {
  return text.replace(/:([A-Za-z0-9_\-\+]+?):/g, function (emojiString) {
    return emoji.get(emojiString) + ' ';
  });
}
```

But I need to add an option for using it, add some tests, and see how it works cross platform.

**Comment by finnp** [comment]
Looks good. Yeah exactly it should not be shown in code blocks. That's the only gotcha.

It's supported by Windows 7+ and Mac. Linux users usually need to change their terminal font. However I think it should be done per default on all systems - and then there should be an option to disable this. I don't think there is a good test if emojis are supported in a terminal.

Maybe you can use this for getting the emojis: https://www.npmjs.com/package/gemoji

**Comment by mikaelbr** [comment]
I used https://www.npmjs.com/package/node-emoji and it seemed to work as expected on the Mac Terminal.

**Comment by mikaelbr** [comment]
Pushed support for emojis now. Would be great if you could test it out and verify.

**Comment by finnp** [comment]
I just tried `:pencil:` and I got `undefined`. So 2 things:

1. It didn't know `:pencil:`, which works here: :pencil:
2. Things like `:thisisbullshit:` should fallback to plain text like this: :thisisbullshit:

**Comment by mikaelbr** [comment]
> Things like :thisisbullshit: should fallback to plain text like this: :thisisbullshit:

Good point. I completely forgot. Fixed now in 9196c03.

> It didn't know :pencil:, which works here: :pencil:

There might be differences, I suspect, but until we find something better I think it'll do. As far as I see there's a pretty [extensive list](https://github.com/omnidan/node-emoji/blob/master/lib/emojifile.js), so hopefully it works for the time being.

**Comment by finnp** [comment]
:pencil: is actually listed here, so might be a bug: https://github.com/omnidan/node-emoji/blob/master/lib/emojifile.js#L561 Will look into this, probably fine to use it.

Otherwise it works fine for me!
